### PR TITLE
Adding VNET Verifier and deployments list to GeoMaster DataProvider

### PR DESCRIPTION
**Pull request by puneetg1983**

### CODE REVIEW

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs [hunk 1]
```diff
@@ -73,6 +73,27 @@ private IGeoMasterClient InitClient()
             return stickyToSlotSettings;
         }
 
+        public async Task<VnetValidationFailureDetails> VerifyHostingEnvironmentVnet(string subscriptionId, string vnetResourceGroup, string vnetName, string vnetSubnetName, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var path = string.Format(@"subscriptions/{0}/providers/Microsoft.Web/verifyHostingEnvironmentVnet", subscriptionId);
```
Comment (hforeste) [inline src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: Do you want to use the SitePathUtility? Just to keep things consistent.

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs [hunk 2]
```diff
@@ -73,6 +73,27 @@ private IGeoMasterClient InitClient()
...
+            var vnetParameters = new VnetParameters { VnetResourceGroup = vnetResourceGroup, VnetName = vnetName, VnetSubnetName = vnetSubnetName };
+            var result = await HttpPost<VnetValidationFailureDetails, VnetParameters>(path, vnetParameters, "", GeoMasterConstants.March2016Version, cancellationToken);
+            return result;
+        }
+
+        public async Task<List<IDictionary<string, dynamic>>> GetDeployments(string subscriptionId, string resourceGroupName, string name)
```
[the remark this answers is not in the document]
Comment (ShekharGupta1988) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: got it. 
An orthogonal question : Are we adding specific methods in geomaster provider and eventually add a pas through ability also?

File: src/Diagnostics.ModelsAndUtils/Models/GeoMaster/VnetParameters.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Models
+{
+    public class VnetParameters
```
Comment (ShekharGupta1988) [inline src/Diagnostics.ModelsAndUtils/Models/GeoMaster/VnetParameters.cs]: I see this is an input model is used only by geomaster data provider
Is this also getting used in csx scripts?

File: src/Diagnostics.ModelsAndUtils/Models/GeoMaster/GeoMasterResponse.cs [hunk 4]
```diff
@@ -14,4 +14,9 @@ public class GeoMasterResponse
         public IDictionary<string, dynamic> Properties { get; set; }
     }
 
+    public class GeoMasterResponseValue
```
Comment (ShekharGupta1988) [inline src/Diagnostics.ModelsAndUtils/Models/GeoMaster/GeoMasterResponse.cs]: same here. Is this class getting used by scripts or just the data provider

File: src/Diagnostics.ModelsAndUtils/Models/GeoMaster/VnetParameters.cs (hunk 3 above)
Comment (hforeste) [reply src/Diagnostics.ModelsAndUtils/Models/GeoMaster/VnetParameters.cs]: No. GeoMasterDataProvider just takes exact string parameters for subscriptionid, vnet subnet name and etc,
@puneetg1983 , you should probably change the access modifier for this model to internal so as to not expose this to developers creating their csx file.

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (hforeste) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: @puneetg1983  and I was talking about that. We can chat offline about this.

[the remark this answers is not in the document]
Comment (puneetg1983) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: This is actually temporary for now. We might come up with a generic pass through method in future that allows any to call a GeoMaster route passing the POST body etc. if required. Some of the these methods *may* be used a lot more often hence I created them.

File: src/Diagnostics.ModelsAndUtils/Models/GeoMaster/VnetParameters.cs (hunk 3 above)
Comment (puneetg1983) [reply src/Diagnostics.ModelsAndUtils/Models/GeoMaster/VnetParameters.cs]: This one I can make internal.

File: src/Diagnostics.ModelsAndUtils/Models/GeoMaster/GeoMasterResponse.cs (hunk 4 above)
Comment (puneetg1983) [reply src/Diagnostics.ModelsAndUtils/Models/GeoMaster/GeoMasterResponse.cs]: Once we add a pass through kind of method, the detectors may benefit directly with this class so I am thinking of leaving this one as is

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (puneetg1983) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: Changed this method to GetAppDeployments

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs (hunk 1 above)
Comment (puneetg1983) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: I don't believe that function can be used directly here as this path has no site name in it...

**Pull request merged**